Bobsar0/Fast-Food-Fast
# #161401830 Consume admin GET and UPDATE orders endpoints

#### What does this PR do?
Enables admin to retrieve all orders and update their status

#### Description of Task to be completed?
- Consume `GET /orders` endpoint so admin can retrieve all orders
- Add search/filter feature to allow admin select/view specific order(s)
- Consume `PUT /orders/:orderId` endpoint so admin can update the status of an order.
- Add color legends to table rows based on order status

#### How should this be manually tested?
- Clone the repo and run the following commands:
```
  - git checkout ft-consume-get-orders-endpoint-#161401830  
  - npm install
  - npm start
```
- Navigate to http://localhost:9999/ and login/create an account using the credentials below
``` 
 username: admin
 email: admin@fastfoodfast.com
 password: admin!23
```
_You should be redirected to a dedicated menu page_
- Navigate to `http://localhost:9999/templates/admin.html`
- Click on **Administrator Panel** > **Manage Orders**
_You should see a table containing the details of all orders_
- Use the search feature at the top of the table to try out the filter option.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/#161401830 

#### Screenshots

![image](https://user-images.githubusercontent.com/19636985/47455561-12963680-d7ca-11e8-9cb2-8ce6856cdacf.png)

![image](https://user-images.githubusercontent.com/19636985/47455652-4f622d80-d7ca-11e8-9351-232106b2c791.png)

![image](https://user-images.githubusercontent.com/19636985/47455738-87697080-d7ca-11e8-9afe-072464d91bd6.png)

![image](https://user-images.githubusercontent.com/19636985/47456121-900e7680-d7cb-11e8-84ef-5939052f9ba7.png)


![image](https://user-images.githubusercontent.com/19636985/47455858-d911fb00-d7ca-11e8-8fa4-6a8be739478b.png)

![image](https://user-images.githubusercontent.com/19636985/47456002-3efe8280-d7cb-11e8-8147-ad99aa0dbd98.png)
